tests/test_operations.py

```diff
@@ -0,0 +1,119 @@
+import random
+
+random.seed(0)
+import numpy as np
+
+np.random.seed(0)
+import torch
+from torch.autograd import grad
+
+torch.manual_seed(0)
+torch.use_deterministic_algorithms(True)
+from pyqtorch.ansatz import AlternateLayerAnsatz
+from pyqtorch.core.batched_operation import batchedRY
+from pyqtorch.core.circuit import QuantumCircuit
+from pyqtorch.core.measurement import total_magnetization
+from pyqtorch.core.operation import CNOT, RY
+from torch.nn import Module, ModuleList
+
+
+class TestFM(QuantumCircuit):
+    def __init__(self, n_qubits: int = 4):
+        super().__init__(n_qubits)
+        self.qubits = range(n_qubits)
+
+    def forward(self, state: torch.Tensor, x: torch.Tensor) -> torch.Tensor:
+        for i in self.qubits:
+            state = RY(x, state, [i], self.n_qubits)
+        return state
+
+
+class TestBatchedFM(QuantumCircuit):
+    def __init__(self, n_qubits: int = 4):
+        super().__init__(n_qubits)
+        self.qubits = range(n_qubits)
+
+    def forward(self, state: torch.Tensor, x: torch.Tensor) -> torch.Tensor:
+        for i in self.qubits:
+            state = batchedRY(x[:, 0], state, [i], self.n_qubits)
+        return state
+
+
+class TestNetwork(Module):
+    def __init__(self, network, n_qubits=4) -> None:
+        super().__init__()
+        self.n_qubits = n_qubits
+        self.network = ModuleList(network)
+        self.state = QuantumCircuit(n_qubits)
```

Comment:
As a minor suggestion: Why not just initiate the state once in the constructor and then reuse it in the forward pass? Something like:
```
def __init__(...):
    ...
    self.init_state = QuantumCircuit.init_state(batch_size, device)
    ...
```

Reply:
but then we'd have to put the batch size in the network definition. 
I don't see the need for that - especially in the tests.